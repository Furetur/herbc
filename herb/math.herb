fn fact(n: int) -> int {
    if n <= 1 {
        return 1;
    }
    return n * fact(n - 1);
}

fn is_prime(x: int) -> bool {
    var i = 2;

    while i * i <= x {
        if x % i == 0 {
            return false;
        }
        i = i + 1;
    }
    return true;
}

fn fib(n: int) -> int {
    var a = 1;
    var b = 1;

    var i = 2;

    while i < 10 {
        var next = a + b;
        a = b;
        b = next;
        i = i + 1;
    }
    return next;
}

fn phi(x: int) -> int {
    var n = x;
	var result = n;

	var i = 2;
	while i * i <= n {
	    if n % i == 0 {
	        while n % i == 0 {
	            n = n / i;
	        }
	        result = result - result / i;
	    }
	    i = i + 1;
	}
	if n > 1 {
	    result = result - result / n;
	}
	return result;
}
