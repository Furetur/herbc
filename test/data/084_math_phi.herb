import herb.math;

entrypoint {
    var i = 1;
    while i < 10000 {
        print(math.phi(i));
        print(",");
        i = i + 100;
    }
}