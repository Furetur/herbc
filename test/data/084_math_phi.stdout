1,100,132,252,400,332,600,700,528,832,720,732,1200,1300,932,1404,1600,972,1800,1900,1232,1900,2100,1392,2058,2400,1632,2592,2800,1932,3000,2652,1920,3300,3204,2328,3312,3700,2160,3772,4000,2732,4200,3520,2916,3852,4452,3132,4800,4368,3332,5100,4452,3240,4900,5500,3732,5700,5800,3360,5632,6100,3744,6300,6192,3920,5280,6700,4532,6732,7000,4716,6804,6216,4932,6912,6900,4800,7504,7900,4536,8100,8004,5532,8100,8500,5520,6720,8112,5544,9000,8604,6132,9100,7488,6332,9600,9504,5940,9900,